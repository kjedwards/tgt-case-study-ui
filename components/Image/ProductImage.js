/* ----------------------------------------------
 * Main product image
 *
 * Author: Kaylee Edwards (z0027hd)
 * Last Modified: 06/08/2018
 * Last Modified By: Kaylee Edwards (z0027hd)
 * --------------------------------------------*/

import React, {Component} from 'react';
import styled, {css} from 'styled-components'


const ImageCont = styled.div`
  display: flex;
  justify-content: center;
  width: 100%;
  height: 60%;
`

const Image = styled.img`
  display: block;
  max-width: 70%;
  max-height: 100%;
`


export default class ProductImage extends Component {
    render () {
        return (
            <ImageCont>
                <Image src={this.props.imageUrl}></Image>
            </ImageCont>
        )
    }
}
