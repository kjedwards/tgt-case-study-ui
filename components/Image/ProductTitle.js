/* ----------------------------------------------
 * Title of product
 *
 * Author: Kaylee Edwards (z0027hd)
 * Last Modified: 06/08/2018
 * Last Modified By: Kaylee Edwards (z0027hd)
 * --------------------------------------------*/

import React, {Component} from 'react';
import styled, {css} from 'styled-components'

const ImageTitle = styled.div`
  color: white;
  font-size: 24px;
  text-align: center;
  padding: 10px;
  width: 80%;
`


export default class ProductTitle extends Component {
    render () {
        return (
            <ImageTitle>This is a placeholder title</ImageTitle>
        )
    }
}
