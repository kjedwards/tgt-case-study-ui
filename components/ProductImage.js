import React, {Component} from 'react';
import styled, {css} from 'styled-components'

import { sizes } from '../constants/DeviceWidth'

const Image = styled.div`
  background-color: black;
  width: 75%;
  height: 60%;

`


export default class ProductImage extends Component {
    render () {
        return (
            <Image></Image>
        )
    }
}
