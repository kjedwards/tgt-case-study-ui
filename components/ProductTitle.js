import React, {Component} from 'react';
import styled, {css} from 'styled-components'

import { sizes } from '../constants/DeviceWidth'

const ImageTitle = styled.div`
  color: white;
  font-size: 24px;
  text-align: center;
  padding: 10px;
  width: 80%;
`


export default class ProductTitle extends Component {
    render () {
        return (
            <ImageTitle>This is a placeholder title</ImageTitle>
        )
    }
}
