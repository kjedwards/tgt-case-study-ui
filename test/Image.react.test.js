import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

//Imported Components
import ProductImage from '../components/Image/ProductImage'
import ProductTitle from '../components/Image/ProductTitle'


describe('ProductImage Component', () => {
  it('RENDERER snapshot', () => {
    const tree = renderer.create(<ProductImage />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('ENZYME snapshot', () => {
    const tree = mount(<ProductImage />)
    expect(toJson(tree)).toMatchSnapshot()
  })
})

describe('ProductTitle Component', () => {
  it('RENDERER snapshot', () => {
    const tree = renderer.create(<ProductTitle />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('ENZYME snapshot', () => {
    const tree = mount(<ProductTitle />)
    expect(toJson(tree)).toMatchSnapshot()
  })
})
