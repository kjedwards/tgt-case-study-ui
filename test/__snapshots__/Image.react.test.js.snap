// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Component ENZYME snapshot 1`] = `
<ProductImage>
  <styled.div>
    <div
      className="sc-bdVaJa eOogis"
    >
      <styled.img>
        <img
          className="sc-bwzfXH canjnE"
        />
      </styled.img>
    </div>
  </styled.div>
</ProductImage>
`;

exports[`ProductImage Component RENDERER snapshot 1`] = `
<div
  className="sc-bdVaJa eOogis"
>
  <img
    className="sc-bwzfXH canjnE"
  />
</div>
`;

exports[`ProductTitle Component ENZYME snapshot 1`] = `
<ProductTitle>
  <styled.div>
    <div
      className="sc-htpNat dtIfKB"
    />
  </styled.div>
</ProductTitle>
`;

exports[`ProductTitle Component RENDERER snapshot 1`] = `
<div
  className="sc-htpNat dtIfKB"
/>
`;
